gateway/operations/putobject.go

```diff
@@ -36,9 +39,7 @@ func (controller *PutObject) RequiredPermissions(_ *http.Request, repoID, _, pat
 	}, nil
 }
 
-func (controller *PutObject) HandleCopy(w http.ResponseWriter, req *http.Request, o *PathOperation, copySource string) {
-	o.Incr("copy_object")
-	// resolve source branch and source path
+func (controller *PutObject) entryForCopyOperation(w http.ResponseWriter, req *http.Request, o *PathOperation, copySource string) *catalog.DBEntry {
```

Comment:
Is it possible the name this one better - it actually answer the request, not just create an entry.
Not part of the 'controller' can be removed (optionally)